src/clj/fluree/db/json_ld/iri.cljc

```diff
@@ -70,6 +68,101 @@
    "_:"                                          24
    f-idx-ns                                      25})
 
+(declare compare-SIDs sid-equiv?)
+
+;; TODO - verify sort order is same!!
+;;
+(deftype SID #?(:clj [^int namespace-code name]
+                :cljs [^number namespace-code name])
+  #?@(:clj  [clojure.lang.Seqable
+             (seq [this] (list (.-namespace-code this) (.-name this)))
+
+             Object
+             (equals [this sid]
+               (sid-equiv? this sid))
+             (hashCode [this] (hash (seq this)))
```

Comment:
I think we could implement this function with `(hash (list namespace-code name))`, then we wouldn't have to implement `Seqable`, unless there's some other reason.

Reply:
I think you are right... I just copied over the same pattern we did for Flake.

Reply:
@zonotope I replaced it with a more optimized hashing and looks like another ~5% txn performance improvement - at least for large transactions.